From 7eb6e77af32658708dad8bab06fce0a6d60f3cb2 Mon Sep 17 00:00:00 2001
From: Jenny TC <jenny.tc@intel.com>
Date: Mon, 15 Dec 2014 15:27:03 +0530
Subject: [PATCH 1/1] max17042: Enable alert after FG config initialization

Initialize interrupt alert and thresholds after initializing
the Fuel Gauge config data.

Change-Id: I43a15358ded6bfa52ddfbb487716c9e44ca15a42
Signed-off-by: Jenny TC <jenny.tc@intel.com>
---
 drivers/power/max17042_battery.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/power/max17042_battery.c b/drivers/power/max17042_battery.c
index bc2989a..db5940d 100644
--- a/drivers/power/max17042_battery.c
+++ b/drivers/power/max17042_battery.c
@@ -918,6 +918,7 @@ static void max17042_init_worker(struct work_struct *work)
 	struct max17042_chip *chip = container_of(work,
 				struct max17042_chip, work);
 	int ret;
+	u32 val;
 
 	/* Initialize registers according to values from the platform data */
 	if (chip->pdata->enable_por_init && chip->pdata->config_data) {
@@ -926,6 +927,14 @@ static void max17042_init_worker(struct work_struct *work)
 			return;
 	}
 
+	/* Enable and Configure IRQs after initializing the FG */
+	if (chip->client->irq) {
+		regmap_read(chip->regmap, MAX17042_CONFIG, &val);
+		val |= CONFIG_ALRT_BIT_ENBL;
+		regmap_write(chip->regmap, MAX17042_CONFIG, val);
+		max17042_set_soc_threshold(chip, 1);
+	}
+
 	chip->init_complete = 1;
 }
 
-- 
1.9.1

